api/src/pcapi/core/users/external/sendinblue.py

```diff
@@ -218,39 +217,22 @@ def make_update_request(payload: UpdateSendinblueContactRequest) -> bool:
 
     try:
         api_instance.create_contact(create_contact)
-        return True
 
     except SendinblueApiException as exception:
-        if exception.status == 524:
-            logger.exception(
-                "Timeout when calling ContactsApi->create_contact",
-                extra={
-                    "email": payload.email,
-                    "attributes": payload.attributes,
-                    "emailBlacklisted": payload.emailBlacklisted,
-                },
-            )
-        else:
-            logger.exception(  # pylint: disable=logging-fstring-interpolation
-                f"Exception when calling ContactsApi->create_contact with status={exception.status}",
-                extra={
-                    "email": payload.email,
-                    "attributes": payload.attributes,
-                    "emailBlacklisted": payload.emailBlacklisted,
-                },
-            )
-        return False
+        if exception.status >= 500:
+            raise ExternalAPIException(is_retryable=True) from exception
 
-    except Exception:  # pylint: disable=broad-except
         logger.exception(
-            "Exception when calling ContactsApi->create_contact",
+            "Exception when calling Sendinblue create_contact API",
             extra={
                 "email": payload.email,
                 "attributes": payload.attributes,
-                "emailBlacklisted": payload.emailBlacklisted,
             },
         )
-        return False
+        raise ExternalAPIException(is_retryable=False) from exception
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** Ici on ne risque pas d'avoir le log en double, avec `logger.exception` + `raise ExternalAPIException(is_retryable=False)` ?
‏

Reply:
Non, les appelants vont normalement catcher cette exception. Dans ce cas, c'est `decorator.py` qui va catcher et qui ne va pas logger.

Mais je suis d'accord que ce point est intéressant : qui a la responsabilité de logger l'erreur ? tel que je le conçois, ce n'est pas forcément le rôle du contrôleur. Mais ça doit dépendre des cas.